code/script.js

```diff
@@ -225,101 +226,133 @@ const generateBoard = () => {
-  // const value =
+  const value = questions.options[questions.selectedIndex].value
 
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
+  console.log('Question selected', currentQuestion)
 }
 
-// This function should be invoked when you click on 'Find Out' button.
-const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
+   // This function invokes when the player click on ‘Find Out’ button. 
+   //Compares the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
+    let count = 0
+    const checkQuestion = () => {
+      count++
 
+      guessCounter.innerHTML = ''
+      guessCounter.innerHTML += `<p>Guess counter: ${count}</p>`
+      const { category, value } = currentQuestion 
+      if (category === 'hair' || category === 'eyes') {
+        if (secret[category] === value) {
+          filterCharacters(true);
+        }
+        else {
+          filterCharacters(false);
+        }
+      }
+      else if (category === 'accessories' || category === 'other') {
+        if (secret[category].includes(value)) {
+        filterCharacters(true);
+      }
+        else {
+        filterCharacters(false);
+      }
+    }
   }
-}
 
-// It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
   if (category === 'accessories') {
     if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
+      alert(`Yes, the person wears ${value}! Keep all people that wears ${value}`)
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
     } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
+      alert(`No, the person doesn't wear ${value}! Remove all people that wears ${value}`)
+        charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
+
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`Yes the person is ${value}! Remove all people that wears ${value}`)
```

Comment:
Instead of 
Remove all people that wears ${value} 
it should be 
Keep all people that wears ${value}
Same for row 310 & 319